lib/varlink.c

```diff
@@ -396,6 +399,26 @@ pwaccess_check_expired(const char *user, long *daysleft, bool *pwchangeable, cha
   if (!user)
     return -EINVAL;
 
+  /*
+   * this comment is regarding all Varlink connections performed in the
+   * context of this PAM module.
+   * the Varlink service will use the UID returned by `sd_varlink_get_peer_uid(link, * &peer_uid);`
+   * for credentials checking.
+   * this UID will be the effective UID of the process seen at connect() time.
```

Comment:
Correct, same behavior as if e.g. login is calling the PAM stack.
Or for the linux kernel if you try to read /etc/shadow with a setuid binary.

In the end this is the reason why we don't relay only on the peer_uid, but also do sanity checks with the parameters and the resulting passwd/shadow struct.
And why we need to take the risk with the setresuid() call.

But with `no_new_privs` this will be different: the effective UID is always the real one, setuid bits are "disabled".
So the peer_uid is always the one of the caller.